NOnion/Network/TorGuard.fs

```diff
@@ -332,14 +342,19 @@ type TorGuard private (client: TcpClient, sslStream: SslStream) =
                             try
                                 do! circuit.HandleIncomingCell cell
                             with
-                            | ex ->
-                                sprintf
-                                    "TorGuard: exception when trying to handle incoming cell type=%i, ex=%s"
-                                    cell.Command
-                                    (ex.ToString())
-                                |> TorLogger.Log
+                            | :? HandshakeFailedException as ex ->
+                                ExceptionUtil.HandleIncomingCellException<HandshakeFailedException>
+                                    cell
+                                    ex
+
+                                self.KillChildCircuits()
+                            | :? CircuitDecryptionFailedException as ex ->
+                                ExceptionUtil.HandleIncomingCellException<CircuitDecryptionFailedException>
+                                    cell
+                                    ex
 
                                 self.KillChildCircuits()
```

Comment:
Should we maybe move this into HandleIncomingCell... as well?

Reply:
I thaught since TorGuard is below this and we use it in TorGuard we couldn't do this 🤔. I tried passing a TorGurad instance but I got errors.